cypress/integration/updateUser.spec.js

```diff
@@ -0,0 +1,731 @@
+          username: '',
+        }
+
+        return userbase.updateUser(updateInfo).then(() => {
+          expect(true, 'updateUser should not be successful').to.be.false
+        }).catch(error => {
+          expect(error).to.be.a('Error')
+          expect(error.name).to.be.equal('UsernameCannotBeBlank')
+        }).finally(() => {
+          return userbase.deleteUser().then(() => {})
+        })
+      })
+    })
+  })
+
+  it('Update user\'s info, UsernameTooLong', function () {
+    cy.window().then((window) => {
+      const { userbase } = window
+      window._userbaseEndpoint = info.endpoint
+      userbase.init({ appId: info.appId })
+
+      return userbase.signUp(randomInfo).then((user) => {
+        cy.log(user)
+        console.log(user)
+
+        const updateInfo = {
+          username: 'a'.repeat(101),
+        }
+
+        return userbase.updateUser(updateInfo).then(() => {
+          expect(true, 'updateUser should not be successful').to.be.false
+        }).catch(error => {
+          expect(error).to.be.a('Error')
+          expect(error.name).to.be.equal('UsernameTooLong')
+        }).finally(() => {
+          return userbase.deleteUser().then(() => {})
+        })
+      })
+    })
+  })
+
+  it('Update user\'s info, CurrentPasswordMissing', function () {
+    cy.window().then((window) => {
+      const { userbase } = window
+      window._userbaseEndpoint = info.endpoint
+      userbase.init({ appId: info.appId })
+
+      return userbase.signUp(randomInfo).then((user) => {
+        cy.log(user)
+        console.log(user)
+
+        const updateInfo = {
+          newPassword,
+        }
+
+        return userbase.updateUser(updateInfo).then(() => {
+          expect(true, 'updateUser should not be successful').to.be.false
+        }).catch(error => {
+          expect(error).to.be.a('Error')
+          expect(error.name).to.be.equal('CurrentPasswordMissing')
+        }).finally(() => {
+          return userbase.deleteUser().then(() => {})
+        })
+      })
+    })
+  })
+
+  it('Update user\'s info, CurrentPasswordIncorrect', function () {
+    cy.window().then((window) => {
+      const { userbase } = window
+      window._userbaseEndpoint = info.endpoint
+      userbase.init({ appId: info.appId })
+
+      return userbase.signUp(randomInfo).then((user) => {
+        cy.log(user)
+        console.log(user)
+
+        const updateInfo = {
+          currentPassword: 'incorrectpassword',
+          newPassword,
+        }
+
+        return userbase.updateUser(updateInfo).then(() => {
+          expect(true, 'updateUser should not be successful').to.be.false
+        }).catch(error => {
+          expect(error).to.be.a('Error')
+          expect(error.name).to.be.equal('CurrentPasswordIncorrect')
+        }).finally(() => {
+          return userbase.deleteUser().then(() => {})
+        })
+      })
+    })
+  })
+
+  it('Update user\'s info, PasswordAttemptLimitExceeded', function () {
```

Comment:
This can be triggered by attempting to use the incorrect password 25 times in a row